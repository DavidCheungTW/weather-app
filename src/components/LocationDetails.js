import React from "react";
import PropTypes from "prop-types";

function LocationDetails(props) {
  const { city, country, errorMessage } = props;
  // if (errorMessage) {
  //   return <h1 className="location-details">{errorMessage}</h1>;
  // }
  // return <h1 className="location-details">{`${city}, ${country}`}</h1>;

  return errorMessage ? (
    <h1>{errorMessage}</h1>
  ) : (
    <h1 className="location-details">{`${city}, ${country}`}</h1>
  );
}

export default LocationDetails;

LocationDetails.defaultProps = {
  errorMessage: "",
};

LocationDetails.propTypes = {
  city: PropTypes.string.isRequired,
  country: PropTypes.string.isRequired,
  errorMessage: PropTypes.string,
};
