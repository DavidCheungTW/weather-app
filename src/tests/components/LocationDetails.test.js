import React from "react";
import { render } from "@testing-library/react";
// import LocationDetails from "../components/LocationDetails";     << platform is incorrect
import LocationDetails from "../../components/LocationDetails";

describe("LocationDetails", () => {
  it("renders the correct city and location props", () => {
    const { getByText } = render(
      <LocationDetails city="Manchester" country="UK" />
    );
    expect(getByText("Manchester, UK")).toBeTruthy(); //  if the text exists as expected with toBeTruthy()
    expect(getByText("Manchester, UK")).toBeInstanceOf(HTMLHeadingElement); //  more specific with our tests
    expect(getByText("Manchester, UK")).toHaveClass("location-details");
  });
});
