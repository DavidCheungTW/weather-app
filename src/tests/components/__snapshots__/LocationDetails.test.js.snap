// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationDetails renders correctly 1`] = `
<DocumentFragment>
  <h1
    class="location-details"
  >
    Manchester, UK
  </h1>
</DocumentFragment>
`;
